pkg/cli/admin/upgrade/status/controlplane.go

```diff
@@ -0,0 +1,78 @@
+package status
+
+import (
+	"io"
+	"text/template"
+
+	v1 "github.com/openshift/api/config/v1"
+)
+
+type assessmentState string
+
+const (
+	assessmentStateProgressing assessmentState = "Progressing"
+	assessmentStateCompleted   assessmentState = "Completed"
+)
+
+type operators struct {
+	Total       int
+	Available   int
+	Progressing int
+	Degraded    int
+}
+
+type controlPlaneStatusDisplayData struct {
+	Assessment assessmentState
+	Completion float64
+	Operators  operators
+}
+
+func assessControlPlaneStatus(cv *v1.ClusterVersion, operators []v1.ClusterOperator) controlPlaneStatusDisplayData {
+	var displayData controlPlaneStatusDisplayData
+	var completed int
+
+	for _, operator := range operators {
+		displayData.Operators.Total++
+		for _, condition := range operator.Status.Conditions {
+			switch {
+			case condition.Type == v1.OperatorAvailable:
+				if condition.Status == v1.ConditionTrue {
+					displayData.Operators.Available++
+				}
+			case condition.Type == v1.OperatorProgressing && condition.Status == v1.ConditionTrue:
+				displayData.Operators.Progressing++
```

Comment:
I doubt `Progressing` is much use, because some ClusterOperator go `Progressing=True` on unrelated-to-update events like the cluster adding a compute node.  But :shrug:, also only a few digits in the output, so I will survive if other folks think it adds value to display this count.

Reply:
I like `Progressing` as an indicator of "something is moving" which I think users generally like - "Progressing: 0" is something that I would hope to avoid in a healthy upgrade.

I agree about imprecision - I was thinking about a followup card about how exactly we would need to compute this.

I personally dislike the `Available` bit - this number is not interesting in most cases, until something goes unavailable, and *then* I think the amount of unavailable ones (1? 10?) is more interesting than Available getting lower.

Reply:
> "Progressing: 0" is something that I would hope to avoid in a healthy upgrade.

Does this happen?  I guess there could be times when the CVO is working with non-ClusterOperator resources where we are updating but have no ClusterOperator attempting to move.  But in that case, I'd want the `status` command to be talking to me about what the CVO was stuck on, not waiting for me to guess from `Progressing: 0`  that it was stuck before I started to do my own digging.

> I personally dislike the Available bit - this number is not interesting in most cases, until something goes unavailable, and then I think the amount of unavailable ones (1? 10?) is more interesting than Available getting lower.

Good point.

Reply:
1. The current iteration only counts a cluster operator as progressing if it has `Progressing=True` and is still at old version. I think that's good enough? I still think it is useful as an indicator that something is moving, so I think we should keep it, at least for now.
2. I kept even `Available: X` for now, but I really think we should present this better, it is mostly just clutter.
3. We actually spend a LOT of time in `Progressing: 0` state, check out the traces in https://docs.google.com/document/d/1rWKPA0DrBU6aA4fsolEfOJIrenzeO5kVKtCoDsMum50/edit?usp=sharing. It seems that most COs will bump themselves right away and don't spend enough time in that state?